File: order-service/src/main/java/com/dtit/order/model/Order.java
package com.dtit.order.model;


import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "orders")
@Data
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_email", nullable = false)
    private String userEmail;

    private BigDecimal total;
    private String status;

    @Column(name = "payment_method")
    private String paymentMethod;

    @Column(name = "tracking_number")
    private String trackingNumber;

    private BigDecimal shipping;

    @Column(name = "created_at")
    private Instant createdAt = Instant.now();

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderItem> items;

}

File: shopping-frontend/app/types/order.ts
export interface OrderItem {
    name: string;
    quantity: number;
    price: number;
    image: string;
}

export interface Order {
    id: string;
    createdAt: string;
    total: number;
    status: string;
    paymentMethod: string;
    trackingNumber?: string;
    shipping: number;
    items: OrderItem[];
}

export type OrderStatus = 'Delivered' | 'Shipped' | 'Processing' | 'Pending'; 